libs/ui/src/smart_cards_screen.tsx

```diff
@@ -554,21 +562,53 @@ export function SmartCardsScreen({
   apiClient,
 }: SmartCardsScreenProps): JSX.Element | null {
   const { programmableCard: card } = auth;
+  const [actionResult, setActionResult] = useState<SmartCardActionResult>();
+
+  useEffect(() => {
+    // Preemptively removing a card that's being programmed can result in a flash of an error
+    // message. To ensure that the error message can be read, continue to display it on the insert
+    // card screen for a few seconds.
+    if (card.status === 'no_card') {
+      const timeout = setTimeout(() => {
+        setActionResult(undefined);
+      }, 3000);
+      return () => clearTimeout(timeout);
+    }
```

Comment:
I could see a case for removing this `setTimeout` and just letting the error persist indefinitely on the insert card screen, until a card is reinserted. Went a bit back and forth here to be honest. Decided not to split hairs since this should all be exceedingly rare in production and is mostly just a response to a cert discrepancy.